larray/core/axis.py

```diff
@@ -1201,6 +1205,11 @@ def align(self, other, join='outer'):
             if not isinstance(other, Axis):
                 other = Axis(other)
             return other
+        elif join == 'exact':
+            if not self.equals(other):
+                raise ValueError("{!r} is not equal to {!r}".format(self, other))
```

Comment:
I think the error message should be more precise, ie it should contain enough information to determine that we were trying to do align with join="exact". This will be especially important to understand what happens when .align will be called by another function. I don't think it is the case now but it will in the future.